### PR TITLE
Fixes #146: on macOS: std.json.JSONException@std/json.d(155): JSONVal…

### LINKED ISSUE #146: on macOS: std.json.JSONException@std/json.d(155): JSONValue is not a string
in dub.json:

```
{
	"name" : "d_test007",
	"description" : "Hello World - A minimal DUB bundle.",
	"dependencies" : {
		"vibe-core": "1.6.2"
	}
}
```

in src/app.d:

```
import vibe.core.log;

void main()
{
}

```
Compile and run it, output:
**std.json.JSONException@std/json.d(155): JSONValue is not a string**

DMD: 2.084.0, 2.085.0, 2.085.1
OS: MacOS 10.14.3

### PR DESCRIPTION
**Pull request by shove70**
…ue is not a string

### CODE REVIEW

File: source/vibe/core/args.d [hunk 1]
```diff
@@ -304,7 +304,7 @@ private T fromValue(T)(in JSONValue val)
 	static if (is(T == bool)) return val.type == JSON_TYPE.TRUE;
 	else static if (is(T : long)) return val.integer.to!T;
 	else static if (is(T : double)) return val.floating.to!T;
-	else static if (is(T == string)) return val.str;
+	else static if (is(T == string)) return (val.type == JSONType.string) ? val.str : val.to!T;
```
Comment (wilzbach) [inline source/vibe/core/args.d]: Isn't it JSON_TYPE?

Comment (wilzbach) [reply source/vibe/core/args.d]: Have a look at https://travis-ci.org/vibe-d/vibe-core/jobs/520577020 (this is with the very old 2.078.3)

```
source/vibe/core/args.d(307,55): Error: undefined identifier JSONType
```

Comment (wilzbach) [reply source/vibe/core/args.d]: As mentioned before, this is due to this change: https://github.com/dlang/phobos/pull/6649

Comment (shove70) [reply source/vibe/core/args.d]: Thanks.

**Pull request closed without merging**